src/bin/check-engine.js

```diff
@@ -0,0 +1,40 @@
+import semver from "semver"
+import pkg from "../../package.json"
+import { execSync } from "child_process"
+import colors from "chalk"
+
+module.exports = function() {
+  const requirements = pkg.engines
+  const nodeVersion = process.env.TESTING
+    ? process.env.FAKE_NODE_VERSION
+    : process.version
+
+  const npm = /^win/.test(process.platform) ? "npm.cmd" : "npm"
+  const npmVersion = process.env.TESTING ?
+    process.env.FAKE_NPM_VERSION :
+    execSync(npm + " --version").toString().trim()
+
+  if (!(
+    semver.satisfies(nodeVersion, requirements.node) &&
+    semver.satisfies(npmVersion, requirements.npm)
+  )) {
+    const errorMessage = colors.yellow(
+      "\n⚠️ " + "Phenomic requires at least " +
+      "node@" + requirements.node + " and npm@" + requirements.npm +
+      "\n\n" +
+      "Your node version is " + nodeVersion +
+      " and your npm version is " + npmVersion +
+      "\n\n" +
+      colors.yellow("See 'Setup' instruction in documentation.") +
+      " " +
+      "https://phenomic.io/docs/setup/"
+    )
+
+    if (process.env.TESTING) {
+      throw new Error(errorMessage)
+    }
+
+    console.log(errorMessage)
+    process.exit(0)
```

Comment:
We have a pretty error message above. I even have a alias like this: `alias npm="npm -s"` to disable that error message. But yeah. exit with 0 sucks since it doesn't stop on CI.